apps/e2e-test/src/steps/app-specific/workflow/workflow.utils.ts

```diff
@@ -0,0 +1,169 @@
+import { ScenarioWorld } from '../../../hooks/ScenarioWorld';
+import { logger } from '../../../logger';
+import { selectSettingDropdown } from '../../../playwright-support/app-specific/element-action';
+import { verifyDeviceStatus } from '../../../playwright-support/app-specific/element-verification';
+import { click } from '../../../playwright-support/generic/element-action';
+import {
+  verifyElementContainsText,
+  verifyElementContainsValue,
+  verifyElementMatchText,
+  verifyElementState,
+  verifyElementText,
+  verifyElementValue,
+} from '../../../playwright-support/generic/element-verification';
+import { waitFor } from '../../../playwright-support/generic/element-wait';
+import { verifyTextMatch, verifyText as verifyTextEqual } from '../../../playwright-support/generic/verification';
+import { TargetSelector } from '../../../utils/selector-mapper';
+import { State, Status } from '../../../utils/types';
+
+export const validateState = async (
+  scWorld: ScenarioWorld,
+  targetSelector: TargetSelector,
+  expState: string,
+  elementIndex?: number | undefined
+) => {
+  for (const state of expState.split('|')) {
+    await waitFor(async () => {
+      await verifyElementState(scWorld.currentPage, targetSelector, state as State, false, elementIndex);
+    });
+  }
+};
+
+export const validateStatus = async (
+  scWorld: ScenarioWorld,
+  targetSelector: TargetSelector,
+  expStatus: Status,
+  elementIndex?: number | undefined
+) => {
+  await waitFor(async () => {
+    await verifyDeviceStatus(scWorld.currentPage, targetSelector, expStatus, false, elementIndex);
+  });
+};
+
+export const validateText = async (
+  scWorld: ScenarioWorld,
+  targetSelector: TargetSelector,
+  expText: string,
+  elementIndex?: number | undefined
+) => {
+  let verifyMethod;
+
+  switch (true) {
```

Comment:
Minor, but I would use `if` rather than `switch`&`case` here which is easier to read and maintain. Switch is not appropriate here.

Reply:
Fixed.